support/linting/src/forbid_as_primitive.rs

```diff
@@ -0,0 +1,78 @@
+use super::*;
+use syn::{visit::Visit, ExprMethodCall, File, Ident};
+
+pub struct ForbidAsPrimitiveConversion;
+
+impl Lint for ForbidAsPrimitiveConversion {
+    fn lint(source: &File) -> Result {
+        let mut visitor = AsPrimitiveVisitor::default();
+
+        visitor.visit_file(source);
+
+        if !visitor.errors.is_empty() {
+            return Err(visitor.errors);
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Default)]
+struct AsPrimitiveVisitor {
+    errors: Vec<syn::Error>,
+}
+
+impl<'ast> Visit<'ast> for AsPrimitiveVisitor {
+    fn visit_expr_method_call(&mut self, node: &'ast ExprMethodCall) {
+        if is_as_primitive(&node.method) {
+            self.errors.push(syn::Error::new(
+                node.method.span(),
+                "Using 'as_*()' methods is banned to avoid accidental panics. Use `try_into()` instead.",
+            ));
+        }
+
+        syn::visit::visit_expr_method_call(self, node);
+    }
+}
+
+fn is_as_primitive(ident: &Ident) -> bool {
+    match ident.to_string().as_str() {
+        "as_u32" | "as_u64" | "as_u128" | "as_usize" => true,
```

Comment:
maybe `u16` as well, or does that not exist for these?

Reply:
just checked, those types don't have an `as_u16` implemented